clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -9529,6 +9529,12 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     // operations which don't have mapping to OpenCL.DebugInfo.100 spec.
     TranslatorArgs.push_back("-spirv-allow-extra-diexpressions");
     TranslatorArgs.push_back("-spirv-allow-unknown-intrinsics=llvm.genx.");
+    bool CreatingSyclSPIRVFatObj =
+        TCArgs.getLastArg(options::OPT_c) &&
```

Comment:
Is it necessary to check for `-c` explicitly?  Is there a reason why the generated device shouldn't be marked to retain the data if we aren't saving it?

Reply:
It would be better to check the actual phase limits here, as `-S` and probably others do some limiting.  `getFinalPhase(Args)` could be used, then check if the final is not `phases::Link`

Reply:
awesome, thanks

Reply:
fixed in latest commit, thx